### PR TITLE
feat: redirect to home page after user switches another wallet address

### PR DESCRIPTION
**Pull request by zhangjing28**
## Description

* redirect to home page after user switches another wallet address

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [ ] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [ ] Updated relevant documentation in the code
- [X] Test code changes manually, and describe your procedure in a comment on this PR
- [ ] Mark this PR with the correct milestone

## If PR to `main`
- [ ] Update the version number in `package.json` and `src/config/common.json`
- [ ] Update the minimum required signer version number in `src/config/common.json` if applicable
- [ ] Test following the procedure in `docs/manual_test.md`
- [ ] Update `CHANGELOG.md`
- [ ] Make sure that all PR's to `manta-signer` and `sdk` on which this PR depends are merged

### CODE REVIEW

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          another wallet address <br />
+          You can come back to you change back to the current wallet address
+        </p>
+        <p
+          className="flex items-center text-white text-xs pt-4 flex text-opacity-60"
+          onClick={handleClick}>
+          {checked ? (
+            <Icon name="greenCheck" className="mr-2 w-4 h-4" />
+          ) : (
+            <svg
```
Comment (zhangjing28) [inline src/pages/SBTPage/components/AddressChangeNotification/index.tsx:63]: Will change to `Icon` component after merge the latest main branch

File: src/pages/SBTPage/SBTContext/mintContext.tsx [hunk 2]
```diff
@@ -115,6 +119,18 @@ export const MintContextProvider = ({ children }: { children: ReactNode }) => {
...
+  useEffect(() => {
+    if (!addressRef?.current) {
+      addressRef.current = externalAccount?.address;
+      return;
+    }
+    if (externalAccount?.address !== addressRef?.current) {
+      addressRef.current = externalAccount?.address;
+      resetContextData();
+      setCurrentStep(Step.Home);
+    }
```
Comment (Kevingislason) [inline src/pages/SBTPage/SBTContext/mintContext.tsx:131]: You could put this code block in a function for improved readability

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        });
+      }
+    } else {
+      // hack Notification component does not support to close it mannually
+      let dom = document.getElementById('address-change-content');
+      while (dom?.parentElement && dom?.parentElement !== document.body) {
+        dom = dom.parentElement;
+      }
+      dom?.remove();
+    }
```
Comment (Kevingislason) [inline src/pages/SBTPage/components/AddressChangeNotification/index.tsx:26]: You could put this code block in a function for improved readability

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  return (
+    <div
+      className="flex items-start pt-2 cursor-pointer"
+      id="address-change-content">
+      <Icon name="warning" fill="#FFA132" />
+      <div className="flex flex-col ml-4 flex-1">
+        <p className="text-white text-sm">
+          Please note that you will be lead to the home page if you switch to
+          another wallet address <br />
+          You can come back to you change back to the current wallet address
```
Comment (Kevingislason) [inline src/pages/SBTPage/components/AddressChangeNotification/index.tsx:55]: Text should probably be "Please note that you will be led to the home page if you switch to another wallet address. You will not lose your progress if you change back to the current wallet address."

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+}: {
+  updateHintStatus: () => void;
+}) => {
+  const [checked, toggleChecked] = useState(false);
+
+  const handleClick = () => {
+    toggleChecked(!checked);
+    setTimeout(() => {
+      updateHintStatus();
+    });
```
Comment (Kevingislason) [inline src/pages/SBTPage/components/AddressChangeNotification/index.tsx:43]: Just curious, why is setTimeout necessary here?

File: src/pages/SBTPage/SBTContext/mintContext.tsx (hunk 2 above)
Comment (zhangjing28) [reply src/pages/SBTPage/SBTContext/mintContext.tsx:131]: I always forget that, thanks for reminding me again. Will do

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx (hunk 4 above)
Comment (zhangjing28) [reply src/pages/SBTPage/components/AddressChangeNotification/index.tsx:55]: That's great help

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx (hunk 5 above)
Comment (zhangjing28) [reply src/pages/SBTPage/components/AddressChangeNotification/index.tsx:43]: This `handleClick` function will do two things after clicking. One is to show `greenCheck` icon and the other one is to close the notification. If without `setTimeout`, the user will only see the notification closed, won't see the `greenCheck` icon.

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx (hunk 3 above)
Comment (zhangjing28) [reply src/pages/SBTPage/components/AddressChangeNotification/index.tsx:26]: Will do

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx (hunk 5 above)
Comment (jincan39) [reply src/pages/SBTPage/components/AddressChangeNotification/index.tsx:49]: so it means setStateWithCallback

File: src/pages/SBTPage/SBTContext/index.tsx [hunk 6]
```diff
@@ -148,6 +151,33 @@ export const SBTContextProvider = (props: { children: ReactElement }) => {
...
+    getHintStatus();
+  }, [config.SBT_NODE_SERVICE, currentStep, externalAccount?.address]);
+
+  const updateHintStatus = useCallback(() => {
+    if (externalAccount?.address) {
+      const url = `${config.SBT_NODE_SERVICE}/npo/hints`;
+      const data = {
+        address: externalAccount.address
+      };
+      axios.post<{ status: boolean }>(url, data);
```
Comment (jincan39) [inline src/pages/SBTPage/SBTContext/index.tsx:176]: For user experience currently there's no way to restore back this setting, just forget it if it's as expect.

**Review by jincan39 — APPROVED** [review]
LGTM

**Review by Kevingislason — APPROVED** [review]

File: src/pages/SBTPage/SBTContext/index.tsx (hunk 6 above)
Comment (zhangjing28) [reply src/pages/SBTPage/SBTContext/index.tsx:176]: Yeah，the result of this `post` function is useless by now, but I prefer add the type to know the type of the result.

File: src/pages/SBTPage/components/AddressChangeNotification/index.tsx (hunk 5 above)
Comment (zhangjing28) [reply src/pages/SBTPage/components/AddressChangeNotification/index.tsx:49]: Yeah, is `setStateWithCallback` a new hook supported by react? I never hear about it.

**Pull request merged**